File: packages/lector/test/fixtures/python-reference/app/checkout.py
from contracts.payment import Order, PaymentProcessor, Receipt


def run_checkout(processor: PaymentProcessor, order: Order) -> Receipt:
    return processor.process(order)


def run_checkout_twice(processor: PaymentProcessor, order: Order) -> Receipt:
    run_checkout(processor, order)
    return run_checkout(processor, order)

File: packages/lector/test/fixtures/go-reference/app/checkout.go
package app

import "fixture.lector.invalid/gomod-reference/contracts"

func RunCheckout(processor contracts.PaymentProcessor, order contracts.Order) (contracts.Receipt, error) {
	return processor.Process(order)
}

func RunCheckoutTwice(processor contracts.PaymentProcessor, order contracts.Order) (contracts.Receipt, error) {
	if _, err := RunCheckout(processor, order); err != nil {
		return contracts.Receipt{}, err
	}
	return RunCheckout(processor, order)
}
